src/components/flyout/flyout.tsx

```diff
@@ -276,19 +281,35 @@ export const EuiFlyout = forwardRef(
 
     // Setting size
     if (isEuiFlyoutSizeNamed(size)) {
-      sizeClassName = sizeToClassNameMap[size];
+      sizeClassName = `euiFlyout--${sizeToClassNameMap[size]}`;
     } else if (newStyle) {
       newStyle.width = size;
     } else {
       newStyle = { ...style, width: size };
     }
 
+    const euiTheme = useEuiTheme();
+    const styles = euiFlyoutStyles(euiTheme, paddingSize);
+
+    const cssStyles = [
+      styles.euiFlyout,
+      styles.euiFlyoutHeader,
+      styles.euiFlyoutBody,
+      styles.euiFlyoutFooter,
+      side === 'left' && type === 'push' && styles['push--left'],
+      isEuiFlyoutSizeNamed(size) && styles[`flyoutSize--${size}`],
+      styles[type],
+      styles[side],
+    ];
+
     const classes = classnames(
       'euiFlyout',
-      typeToClassNameMap[type as _EuiFlyoutType],
-      sideToClassNameMap[side as _EuiFlyoutSide],
+      {
+        [`euiFlyout--${type}`]: type,
+        [`euiFlyout--${side}`]: side,
+        [`euiFlyout--padding-${paddingSize}`]: paddingSize,
+      },
       sizeClassName,
-      paddingSizeToClassNameMap[paddingSize as _EuiFlyoutPaddingSize],
       widthClassName,
       className
     );
```

Comment:
We shouldn't need to apply any of these modifier classes to EuiFlyout any longer

```suggestion
    const classes = classnames('euiFlyout', className);
```